src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -439,13 +441,22 @@ void TestAnnotations(ImmutableArray<TextSpan> expectedSpans, string annotationKi
         private static Document GetDocumentToVerify(DocumentId expectedChangedDocumentId, Solution oldSolution, Solution newSolution)
         {
             Document document;
-            // If the expectedChangedDocumentId is not mentioned then we expect only single document to be changed
+
             if (expectedChangedDocumentId == null)
             {
                 var projectDifferences = SolutionUtilities.GetSingleChangedProjectChanges(oldSolution, newSolution);
+
                 var documentId = projectDifferences.GetChangedDocuments().FirstOrDefault() ?? projectDifferences.GetAddedDocuments().FirstOrDefault();
-                Assert.NotNull(documentId);
-                document = newSolution.GetDocument(documentId);
+                if (documentId == null)
+                {
+                    // pick a random document if no document changed
```

Comment:
i don't understand what's going on here.

Reply:
In the scenario of no document changes, still need to pick a document to continue with the rest of test.

Reply:
That seems pretty hacky.  I would not expect to even go through this codepath for the upgrade project tests.

Reply:
Agreed.